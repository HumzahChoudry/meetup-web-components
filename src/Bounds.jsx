import React from 'react';
import cx from 'classnames';

export const BOUNDS_CLASS = 'bounds';

/**
 * Design System Component: Provides `bounds` container for components
 * @module Bounds
 */
class Bounds extends React.Component {
	render() {
		const {
			children,
			className,
			...other
		} = this.props;

		const classNames = cx(
			BOUNDS_CLASS,
			className
		);

		return (
			<div
				className={classNames}
				{...other}>
					{children}
			</div>
		);
	}
}

export default Bounds;
