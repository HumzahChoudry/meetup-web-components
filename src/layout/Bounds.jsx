import PropTypes from 'prop-types';
import React from 'react';
import cx from 'classnames';

import withLoading from '../utils/components/withLoading';

export const BOUNDS_CLASS = 'bounds';

/**
 * Design System Component: Provides `bounds` container for components
 * @module Bounds
 */
export class Bounds extends React.Component {
	render() {
		const {
			children,
			className,
			narrow,
			loadingProps = {}, // eslint-disable-line no-unused-vars
			isLoading,
			...other
		} = this.props;

		const classNames = cx(
			BOUNDS_CLASS,
			{
				'bounds--wide': !narrow,
				'component--isLoading': isLoading
			},
			className
		);

		return (
			<div
				className={classNames}
				{...other}
			>
				{children}
			</div>
		);
	}
}
Bounds.propTypes = {
	narrow: PropTypes.bool,
	isLoading: PropTypes.bool,
	loadingProps: PropTypes.shape({
		color: PropTypes.string,
		scrimColor: PropTypes.string,
		size: PropTypes.string
	})
};

export default withLoading(Bounds);
