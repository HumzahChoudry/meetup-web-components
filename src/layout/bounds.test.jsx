import React from 'react';
import { shallow } from 'enzyme';

import { Bounds } from './Bounds';

const WIDE_CLASS = 'bounds--wide';

describe('Bounds', function() {
	const bounds = shallow(<Bounds />);

	it('exists', function() {
		expect(bounds).toMatchSnapshot();
	});
	it(`check that default component has '${WIDE_CLASS}' class`, function() {
		expect(bounds.find(`.${WIDE_CLASS}`).length).not.toBe(0);
	});
	it("check that narrow component does not have the 'bounds--wide' class", function() {
		const boundsNarrow = shallow(<Bounds narrow />);
		expect(boundsNarrow.find(`.${WIDE_CLASS}`).length).toBe(0);
	});
});
