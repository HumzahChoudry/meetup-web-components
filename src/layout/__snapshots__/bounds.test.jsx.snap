// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bounds exists 1`] = `
<div
  className="bounds bounds--wide"
/>
`;
